src/main/java/horus/tasks/Task.java

```diff
@@ -0,0 +1,92 @@
+package horus.tasks;
+
+/**
+ * An object representing a task
+ */
+public class Task {
+    String taskDesc;
+    boolean isMarked;
+
+    /**
+     * Creates a task
+     **
+     * @param taskStr String representing the description of task to be created
+     */
+    public Task(String taskStr) {
+        this.taskDesc = taskStr;
+        this.isMarked = false;
+    }
+
+    /**
+     * Returns string to be saved to file
+     *
+     * @return A string from which a similar Task object may be initialized from
+     */
+    public String getTaskData() {
+        String markStr = this.isMarked? "T" : "F";
+        return markStr + "," + this.taskDesc;
+    }
+
+    /**
+     * Creates a Task from data previously saved to a file with Task.getTaskData()
+     *
+     * @param taskData A string containing task data, similar to one from Task.getTaskData()
+     * @return A Task object whose getTaskData() method will return a string similar to taskData
+     */
+    public static Task readTaskData(String taskData) throws InvalidInputException {
+        Task out;
+        String[] taskDataArray = taskData.split(",", 3);
+        String taskType = taskDataArray[0];
+        String markedStr = taskDataArray[1];
+        String taskStr = taskDataArray[2];
+        switch (taskType) {
+        case "T":
+            out = new ToDoTask(taskStr);
+            break;
+        case "D":
+            if(!taskStr.contains("/by")){
```

Comment:
```suggestion
            if(!taskStr.contains("/by")) {
```

You might have missed a space between the ) and {. Small issue though!